src/createApp.ts

```diff
@@ -334,7 +334,10 @@ function resolveFactory(type: FactoryTypes, { factory, instance }: ItemDefinitio
 	}
 
 	if (typeof instance === 'object') {
-		return Promise.resolve(() => instance);
+		// FIXME: Factories can return instances without having to return a promise for them, why does that suddenly
+		// cause TypeScript to throw a fit?!
+		const promise = Promise.resolve(instance);
+		return Promise.resolve(() => promise);
```

Comment:
I'm returning a promise for a factory, which is a function that returns an instance, or a promise for an instance.

Returning a promise for a function that returns an instance no longer works.